wiki/simulation/simulating-vehicle-using-autoware.md

```diff
@@ -0,0 +1,109 @@
+
+![](../../assets/images/team_j_wiki_automodels.png)*Figure 1. Different vehicle models*
+ 
+ 
+### Available Worlds
+Gazebo uses an ‘empty world’  which lacks any structures, to simulate the environment we need a world file. Autoware provides three templates of world files which can be downloaded from [car_sim](http://gazebosim.org/blog/car_sim )
+
+![](../../assets/images/team_j_wiki_autoworlds.png)*Figure 2. Gazebo worlds developed by autoware*
+ 
+All three worlds also have the Point Cloud Maps (PCD files) available for download which are needed for NDT based localization.
+One can always use other gazebo worlds but the PCD maps have to be manually generated and downsampled, the process can be found on [Autoware Documentation](https://readthedocs.org/projects/autoware/downloads/pdf/feature-documentation_rtd/)
+ 
+## Using the path planning algorithms provided in Autoware
+Autoware provides a number of different types of path planning algorithms. These include simpler ones like pure pursuit based on a set of pre-configured waypoints to dynamic path planning using LiDAR and camera based costmaps to a custom software package called OpenPlanner that can use vector maps and sensor data for planning.
+### 1. Waypoint following:
+- A set of waypoints can be generated by using the waypoint_saver node within Autoware, which will record the position, orientation and velocity at a customizable set of intervals. Within the simulation, the vehicle can be moved with teleoperation or other manual approaches, and in the real world, this may be with a joystick or manual control of the vehicle.
+- The previously generated set of waypoints are loaded using the waypoint_loader node within Autoware.
+- The lane_rule, lane_stop and lane_select topics are checked.
+- The astar_avoid is selected (and if not selected, it will not use obstacle avoidance)
+- The velocity_set is checked (configured for a fixed velocity between waypoints)
+- The twist_filter node is checked (to allow for limiting of the resulting values for control of the vehicle)
+- The vehicle interface nodes are selected (for control of the vehicle)
+- The pure_pursuit node is checked (and at this point, the vehicle will start moving)
+
+A sample image with waypoints is shown below.
+
+ ![](../../assets/images/team_j_wiki_autoplan.png)*Figure 3. Sample waypoints*
+ 
+This shows how the waypoints provide a velocity and orientation at every location, and how pure pursuit plans a path that interpolates them, as well as a lookahead target for the vehicle while driving.
+## 2. Using OpenPlanner for dynamic path planning
```

Comment:
You're going to want to insert a line above the ``##` or it likely won't render correctly.